apps/src/components/map-layers/s2d-variable-values.tsx

```diff
@@ -30,250 +81,799 @@ const tooltipTemperatureRange = __(
-		'instead.'
+			'or below): The accuracy of past forecasts was no better than random ' +
+			'chance, so the forecast should not be used. The historical ' +
+			'climatology is a better guide than the forecast and can be used ' +
+			'instead.'
 	),
 	__(
 		'The skill level at this location is Low (CRPSS value is between ' +
-		'0.00 and 0.05): Past forecasts provided only a small improvement ' +
-		'over random chance. Use these forecasts with caution and consider ' +
-		'consulting both the forecasts and the historical climatology.'
+			'0.00 and 0.05): Past forecasts provided only a small improvement ' +
+			'over random chance. Use these forecasts with caution and consider ' +
+			'consulting both the forecasts and the historical climatology.'
 	),
 	__(
 		'The skill level at this location is Medium (CRPSS value is between ' +
-		'0.05 and 0.25): The accuracy of past forecasts was satisfactory. ' +
-		'The forecast is a better guide than the historical climatology.'
+			'0.05 and 0.25): The accuracy of past forecasts was satisfactory. ' +
+			'The forecast is a better guide than the historical climatology.'
 	),
 	__(
 		'The skill level at this location is High (CRPSS value is above ' +
-		'0.25): Past forecasts were mostly accurate. The forecast is ' +
-		'considered trustworthy.'
+			'0.25): Past forecasts were mostly accurate. The forecast is ' +
+			'considered trustworthy.'
 	),
 ];
 
-export const S2DVariableValues: React.FC<S2DVariableValuesComponentProps> = ({
-	dateRangeYears,
-	historicalMedian,
-	nearNormalTemperatureRange,
-	skill,
-}) => {
+/**
+ * Generate a period range label for a given date range and frequency.
+ *
+ * @param dateRangeStart - Start date of the period. Example: 2025-08-01
+ * @param frequency - Frequency type
+ * @param locale - Locale to use for formatting
+ */
+const generatePeriodRangeLabel = (
+	dateRangeStart: string,
+	frequency: S2DFrequencyType,
+	locale: string
+): string | null => {
+	const periodStart = utc(dateRangeStart);
+
+	if (!periodStart) {
+		return null;
+	}
+
+	const periodStartLabel = Intl.DateTimeFormat(locale, {
+		month: 'long',
+		timeZone: 'UTC',
+	}).format(periodStart);
+
+	if (frequency === FrequencyType.MONTHLY) {
+		return periodStartLabel;
+	}
+
+	const periodEnd = getPeriodEnd(periodStart, frequency);
+	const periodEndLabel = Intl.DateTimeFormat(locale, {
+		month: 'long',
+		timeZone: 'UTC',
+	}).format(periodEnd);
+
+	return sprintf(__('%s to %s'), periodStartLabel, periodEndLabel);
+};
+
+/**
+ * Return true if two numbers are in the same thousand.
+ */
+const isSameThousand = (valueA: number, valueB: number): boolean =>
+	Math.floor(valueA / 1000) === Math.floor(valueB / 1000);
+
+/**
+ * Return true if the given RGB colour code is white.
+ *
+ * @param colour - The colour code, prefixed with `#`.
+ */
+const isWhite = (colour: `#${string}`): boolean =>
+	colour.toUpperCase() === '#FFFFFF' || colour.toUpperCase() === '#FFF';
```

Comment:
```suggestion
	colour.toUpperCase().startsWith('#FFF');
```

That should cover both situations.